### ISSUE TITLE
Add a Share / link button on location page

### ISSUE THREAD

**Issue by wbazant** [falling-fruit/falling-fruit-web#636]
Let's help Falling Fruit to spread through our users' social graphs by providing an easy way to share locations.

I think the upper right corner would be the place, instead of the arrow down that reveals links to Wikipedia / Eat The Weeds:

![Screenshot from 2024-12-10 13-27-47](https://github.com/user-attachments/assets/cdb66d76-7474-4524-9b10-4b9146ed1ed5)

I suggest removing the links and letting users rely on the Latin name to find out about the species: when I was watching a friend use a site, he at some point selected the text and paste it into google when he wanted to learn more about an unknown plant.

We also had a suggestion to place them just as glyph icons immediately by the type name - but I think each additional visual element detracts from the whole, while the value of a wikipedia link seems limited over having the user google the name.

I propose to share clean URLs to locations, e.g. beta.fallingfruit.org/locations/1809488 - without the map location - since they look nice in whatsapp chats.

The button should look like 🔗 and I'm not sure about the interaction with it. The ones I've seen online are:
- either pressing the button should copy directly to clipboard and toast the message about it
- or when clicked it should open a textarea with a link text, as well as a second icon which represents copying to clipboard

**Comment by ezwelty** [comment]
> the value of a wikipedia link seems limited over having the user google the name

@wbazant I have to disagree with this statement, since the links associated with each type are carefully curated and link directly to good resources about that particular type.

However, as you know, I support dropping the space-inefficient type accordions in favor of something like what we have on the mobile list. I hear you about not filling the location page with url logos. So I'm okay dropping the links for now in favor of other functionality, if in the future clicking on a type goes to a type profile page (with the links and much more!) as per the long-term vision of more interactive types:

https://docs.google.com/document/d/1ah6jKO9uizBqeBtTVoIXi51gpYEQEyYzzastcMKOp5Y/edit?tab=t.0#heading=h.cbt21puyoh05

As for the shareable URL, I like the idea, the proposed icon (boxicon `link`), and the idea of it being copied directly to the clipboard with a toast. 🥳 (Could a link icon, with a second offset link icon, combine the best of link and copy-to-clipboard icon conventions?)

I'll note that this is related to #113, since ideally the link preview displayed by the social media platform would reflect the location, not the website as a whole.

**Comment by wbazant** [comment]
This would probably be one UI change with #530 (which is about the types taking too much space as they are now)

**Comment by wbazant** [comment]
My apple friend just shared a list of apple trees in Glasgow to inspire people to go prune them if they want to practice, and linked his Google maps page and old Falling Fruit page. Sharing views of the site is definitely a use case and it would be nice to consistently let users do it. 


Shall we make the "share this view" part of a search + filter ribbon, and offer across the map, list, and location drawer? There can be a URL to copy, and maybe checkboxes about what state to include in the URL.

**Comment by ezwelty** [comment]
@wbazant On the existing website, a "Link" link above the map and in the location infowindow updates the browser URL to reflect the current state, which can then be selected and copied. If I understand correctly, we get this functionality for free now on desktop, because the URL is always updated, but it is clunkier to access the URL on mobile, hence the need for such a button? 

> maybe checkboxes about what state to include in the URL

I think I prefer the original simpler suggestion of copy-to-clipboard and toast, because most users will want the link to reproduce the current state.

There's overlap here with high-priority #517.

**Comment by wbazant** [comment]
It's to make it easy, but also discoverable! The URL only keeps some state - part of the site you're on, map view, id of location being viewed, etc. It could be made to work - e.g. changing a filter would push a change to the URL - but it's not clear on a typical page whether sharing a long messy URL with another person will let them view the same thing. Plus it's clunkier in a mobile browser and wouldn't work on the mobile app at all.

> copy-to-clipboard and toast

If it's a button next to the filter button then I'm imagining it could open a similar overlay, with a textarea containing some words that explain this is about sharing - or summary of what is being shared - and then a textarea containing the link and a little copy symbol next to it, that toasts after clicking.

I would want this to support #517 and also help understand that the data on Falling Fruit is to be shared socially, and encourage sending those links.

**Comment by wbazant** [comment]
Technical challenge is how to pack a selection from a type filter into a string, we have ways of compressing the choices:
- ranges like (1,10)
- nesting in the tree
- categories like "default" or "freegan" or "all"
